src/managers/ChannelManager.js

```diff
@@ -22,10 +22,10 @@ class ChannelManager extends BaseManager {
   add(data, guild, cache = true) {
     const existing = this.cache.get(data.id);
     if (existing) {
-      if (existing._patch && cache) existing._patch(data);
-      if (guild) guild.channels?.add(existing);
-      if (ThreadChannelTypes.includes(existing.type) && typeof existing.parent?.threads !== 'undefined') {
-        existing.parent.threads.add(existing);
+      if (cache) existing._patch(data);
+      guild?.channels?.add(existing);
+      if (ThreadChannelTypes.includes(existing.type)) {
+        existing.parent?.threads.add(existing);
```

Comment:
The original code checks for the existence of `threads` too:
```suggestion
        existing.parent?.threads?.add(existing);
```

Unless, of course, that it was to check the parent. cc: @ckohen

Reply:
Theoretically its impossible to have a parent channel that doesn't have a ThreadManager, but in case threads can ever be in other channel types in the future (before we get around to adding ThreadManagers to them) it checks.